### PR TITLE
Feature/11 - Add Foreign Keys

### LINKED ISSUE #11: Schema: Add foreign key constraints
should be in a format like 

```
table.foreign('user_id').references('id').on('users')
```

### PR DESCRIPTION
**Pull request by josephmancuso**
Closes #11 

@Marlysson @sphinxxanxus Refactored and added foreign keys here

- [x] On Delete, Cascade
- [x] Refacter alter grammar
- [x] Add constraints to alter grammar

### CODE REVIEW

File: src/masonite/orm/grammar/BaseGrammar.py [hunk 1]
```diff
@@ -102,6 +105,19 @@ def _compile_alter(self):
         self._sql = sql
```
Comment (corentinalcoy) [inline src/masonite/orm/grammar/BaseGrammar.py:105]: image
It's should be 
```MySQL
alter table article 
add constraint article_user_id_foreign foreign key(`user_id`) references `users`(`id`)
```

Comment (corentinalcoy) [reply src/masonite/orm/grammar/BaseGrammar.py:105]: I think you should update the alter to compile the  foreign constraints

File: src/masonite/orm/blueprint/Blueprint.py [hunk 2]
```diff
@@ -64,13 +64,30 @@ def __init__(self, column, constraint_type):
...
+        self.foreign_column = foreign_column
+        self.index_name = ""
+
+    def references(self, foreign_column):
+        self.foreign_column = foreign_column
+
+    def on(self, foreign_table):
+        self.foreign_table = foreign_table
+        self.index_name = f"{self.current_table}_{self.column_name}_foreign"
+
```
Comment (corentinalcoy) [inline src/masonite/orm/blueprint/Blueprint.py:81]: I think we should add the `on_delete` and `on_update` methods

File: src/masonite/orm/grammar/BaseGrammar.py (hunk 1 above)
Comment (josephmancuso) [reply src/masonite/orm/grammar/BaseGrammar.py:105]: @corentinalcoy yes you are right.

Comment (josephmancuso) [reply src/masonite/orm/grammar/BaseGrammar.py:105]: I'll need to refactor the alter grammar like I did for the create grammar

File: src/masonite/orm/blueprint/Blueprint.py (hunk 2 above)
Comment (josephmancuso) [reply src/masonite/orm/blueprint/Blueprint.py:81]: Yes also right

**Pull request merged**